src/org/thoughtcrime/securesms/PinnedMessageAdapter.java

```diff
@@ -120,9 +120,20 @@ public void onClick(View v) {
     }
 
     private void setMessageView(MessageRecord record, ViewHolder viewHolder) {
-        if(!record.isOutgoing()) {
-            viewHolder.recipient.setText(record.getRecipient().getName());
+        String ownMessageName = "Me";
+
+        if (record.isOutgoing()) {
+            viewHolder.recipient.setText(ownMessageName);
+            return;
         }
+
+        String messageSenderName = record.getRecipient().getName();
+
+        if (messageSenderName == null) {
+            messageSenderName = record.getRecipient().getAddress().toString();
```

Comment:
That's the simplest and most efficient way to complete this story #21

Reply:
looks good!

Reply:
👍